core/src/main/java/cucumber/runtime/formatter/PrettyFormatter.java

```diff
@@ -213,16 +213,19 @@ String formatStepText(String keyword, String stepText, Format textFormat, Format
         for (Argument argument : arguments) {
             // can be null if the argument is missing.
             if (argument.getOffset() != null) {
-                String text = stepText.substring(textStart, argument.getOffset());
-                result.append(textFormat.text(text));
+                int textEnd = argument.getOffset();
+                if (textStart <= textEnd ) { // textStart can be > textEnd with nested capture groups
+                    String text = stepText.substring(textStart, textEnd);
+                    result.append(textFormat.text(text));
+                }
             }
             // val can be null if the argument isn't there, for example @And("(it )?has something")
             if (argument.getVal() != null) {
                 result.append(argFormat.text(argument.getVal()));
                 textStart = argument.getOffset() + argument.getVal().length();
```

Comment:
I think you're missing a bounds check here.

Though I regret suggesting bounds checking in the first place as the solution. Having looked at the problem a bit more closely I think it would be more elegant to skip nested arguments all together. You can skip processing an element in an array using the `continue` statement. 

https://docs.oracle.com/javase/tutorial/java/nutsandbolts/branch.html
https://stackoverflow.com/questions/389741/what-is-the-continue-keyword-and-how-does-it-work-in-java